api/query/cache/monitor/monitor.go

```diff
@@ -114,19 +116,36 @@ func (m *indexMonitor) Start() error {
 		return err
 	}
 
+	m.idx.SetIngestChan(m.runIngested)
+
+	reqs := uint(0)
 	for {
 		if !m.isRunning {
 			return errStopped
 		}
 
-		heapBytes := m.rt.GetHeapBytes()
-		if heapBytes > m.maxHeapBytes {
-			m.logger.Errorf("Out of memory: %d > %d", heapBytes, m.maxHeapBytes)
-			m.idx.EvictRuns(m.percent)
-		} else {
-			m.logger.Debugf("Monitor: %d heap-allocated bytes OK", heapBytes)
+		timer := make(chan bool, 1)
+		go func() {
+			time.Sleep(m.interval)
+			timer <- true
+		}()
+
+		for {
+			select {
+			case <-timer:
+				m.check()
```

Comment:
Move `m.check()` (and the other one in the next `case`) out of this `for` loop.

Reply:
The `m.check()` in the other case is conditional. That's why I don't have it outside the `for` loop.